velox/functions/sparksql/tests/SparkCastExprTest.cpp

```diff
@@ -196,6 +194,11 @@ TEST_F(SparkCastExprTest, invalidDate) {
       {"2015-031-8"},
       "Unable to parse date value: \"2015-031-8\"",
       VARCHAR());
+  testInvalidCast<std::string>(
+      "date",
+      {"-1-1-1"},
```

Comment:
Would you add more tests like -111-1-1 to make sure sign is excluded when doing the digits check?